### PR TITLE
fix: compoudv3 morphoblue config tokens

### PR DESCRIPTION
**Pull request by LuPoYi**
- [x] change compoudv3 morphoblue config tokens - use `Token`, not just address
- [x] yarn changeset

### CODE REVIEW

File: src/logics/compound-v3/configs.ts [hunk 1]
```diff
@@ -31,22 +32,16 @@ export const configs: Config[] = [
...
-        baseTokenAddress: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
+        comet: mainnetTokens.cUSDCv3,
+        baseToken: mainnetTokens.USDC,
       },
       {
         id: MarketId.ETH,
-        cometAddress: '0xA17581A9E3356d9A858b789D68B4d866e593aE94',
-        baseTokenAddress: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
+        comet: mainnetTokens.cWETHv3,
+        baseToken: mainnetTokens.WETH,
```
Comment (Jeff-CCH) [inline src/logics/compound-v3/configs.ts:41]: Actually, I was wondering if we could move common tokens like WETH to the common package to reduce the config size of protocols. What do you think?

File: src/logics/compound-v3/logic.borrow.test.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ describe('CompoundV3 BorrowLogic', function () {
         const { marketId, output } = fields;
         const market = getMarket(chainId, marketId);
 
-        expect(routerLogic.to).to.eq(market.cometAddress);
```
Comment (Jeff-CCH) [inline src/logics/compound-v3/logic.borrow.test.ts:45]: please also help fix the `cometAddress` in test/logics/compound-v3/helpers.ts

File: src/logics/compound-v3/scripts/refresh-tokens-data.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ export default async function () {
     const service = new Service(chainId);
 
     const tokenMap: Record<string, common.Token> = {};
+
```
Comment (Jeff-CCH) [inline src/logics/compound-v3/scripts/refresh-tokens-data.ts:13]: Do we need this blank line?

File: src/logics/compound-v3/tokens/data/arbitrum.json [hunk 4]
```diff
@@ -1,17 +1,38 @@
...
     "decimals": 6,
     "symbol": "USDC",
-    "name": "USD Coin (Arb1)"
+    "name": "USD Coin"
+  },
+  "cUSDCev3": {
+    "chainId": 42161,
+    "address": "0xA5EDBDD9646f8dFF606d7448e414884C7d905dCA",
+    "decimals": 6,
+    "symbol": "cUSDCv3",
```
Comment (Jeff-CCH) [inline src/logics/compound-v3/tokens/data/arbitrum.json:27]: should we use `cUSDCev3`?

File: src/logics/compound-v3/tokens/data/arbitrum.json [hunk 5]
```diff
@@ -1,17 +1,38 @@
...
+    "name": "Compound USDC"
```
Comment (Jeff-CCH) [inline src/logics/compound-v3/tokens/data/arbitrum.json:28]:
```suggestion
    "name": "Compound USDCe"
```

File: src/logics/compound-v3/tokens/data/arbitrum.json [hunk 6]
```diff
@@ -1,17 +1,38 @@
...
+  },
+  "USDC.e": {
+    "chainId": 42161,
+    "address": "0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8",
+    "decimals": 6,
+    "symbol": "(USDC.e)",
```
Comment (Jeff-CCH) [inline src/logics/compound-v3/tokens/data/arbitrum.json:34]: Do we need the round brackets `()`?

File: src/logics/compound-v3/tokens/data/mainnet.json [hunk 7]
```diff
@@ -75,5 +75,12 @@
     "decimals": 18,
     "symbol": "wstETH",
     "name": "Wrapped liquid staked Ether 2.0"
+  },
+  "rETH": {
```
Comment (Jeff-CCH) [inline src/logics/compound-v3/tokens/data/mainnet.json:79]: Is this for lending-sdk?

File: src/logics/compound-v3/tokens/data/polygon.json [hunk 8]
```diff
@@ -40,5 +47,19 @@
...
     "name": "Wrapped Matic"
+  },
+  "MaticX": {
+    "chainId": 137,
+    "address": "0xfa68FB4628DFF1028CFEc22b4162FCcd0d45efb6",
+    "decimals": 18,
+    "symbol": "MaticX",
+    "name": "Liquid Staking Matic (PoS)"
+  },
+  "stMATIC": {
```
Comment (Jeff-CCH) [inline src/logics/compound-v3/tokens/data/polygon.json:58]: I didn't find this token in[ compound-v3 config](https://github.com/compound-finance/comet/blob/main/deployments/polygon/usdc/configuration.json). Where would we use this token for?

File: src/logics/compound-v3/tokens/index.ts [hunk 9]
```diff
@@ -9,3 +12,15 @@ export const mainnetTokens = common.toTokenMap<MainnetTokenSymbols>(mainnetToken
...
+
+export const baseTokens = common.toTokenMap<BaseTokenSymbols>(baseTokensJSON);
+
+type ArbitrumTokenSymbols = keyof typeof arbitrumTokensJSON;
+
+export const arbitrumTokens = common.toTokenMap<ArbitrumTokenSymbols>(arbitrumTokensJSON);
+
+type OptimismTokenSymbols = keyof typeof optimismTokensJSON;
+
+export const optimismTokens = common.toTokenMap<OptimismTokenSymbols>(optimismTokensJSON);
```
Comment (Jeff-CCH) [inline src/logics/compound-v3/tokens/index.ts:26]: Please arrange them in chain id (ascending)

File: src/logics/morphoblue/configs.ts [hunk 10]
```diff
@@ -5,8 +5,8 @@ type ContractNames = 'Morpho' | 'MorphoFlashLoanCallback';
 
 export interface MarketConfig {
   id: string;
-  loanTokenAddress: string;
```
Comment (Jeff-CCH) [inline src/logics/morphoblue/configs.ts:8]: please also fix the `loanTokenAddress` in test/logics/morphoblue/helpers.ts

File: src/logics/morphoblue/configs.ts [hunk 11]
```diff
@@ -5,8 +5,8 @@ type ContractNames = 'Morpho' | 'MorphoFlashLoanCallback';
...
-  collateralTokenAddress: string;
```
Comment (Jeff-CCH) [inline src/logics/morphoblue/configs.ts:9]: same as `loanTokenAddress`

File: src/logics/compound-v3/configs.ts (hunk 1 above)
Comment (LuPoYi) [reply src/logics/compound-v3/configs.ts:41]: Agree the idea. We will build the common token list soon. After that, we can revisit this and update the tokens to use the common token list.

File: src/logics/compound-v3/tokens/index.ts (hunk 9 above)
Comment (LuPoYi) [reply src/logics/compound-v3/tokens/index.ts]: arranged

File: .changeset/hungry-wombats-study.md [hunk 12]
```diff
@@ -0,0 +1,5 @@
+---
+'@protocolink/logics': minor
+---
+
+fix compoudv3 morphoblue config tokens
```
[the remark this answers is not in the document]
Comment (LuPoYi) [reply .changeset/hungry-wombats-study.md]: fixed

File: .changeset/hungry-wombats-study.md [hunk 13]
```diff
@@ -0,0 +1,5 @@
+---
+'@protocolink/logics': minor
```
[the remark this answers is not in the document]
Comment (LuPoYi) [reply .changeset/hungry-wombats-study.md]: indeed, let's change to `minor`

File: src/logics/compound-v3/logic.borrow.test.ts (hunk 2 above)
Comment (LuPoYi) [reply src/logics/compound-v3/logic.borrow.test.ts:45]: fixed, sorry for not notice `test` part

File: src/logics/compound-v3/tokens/data/mainnet.json (hunk 7 above)
Comment (LuPoYi) [reply src/logics/compound-v3/tokens/data/mainnet.json]: Removed. `rETH` is a collateral token built from `refresh-tokens-data.ts`, which we won't be using.

File: src/logics/compound-v3/tokens/data/polygon.json (hunk 8 above)
Comment (LuPoYi) [reply src/logics/compound-v3/tokens/data/polygon.json]: Removed. rETH is a collateral token built from refresh-tokens-data.ts
I think their doc is out of date, you could find it on https://app.compound.finance/?market=usdc.e-polygon and onchain data

File: src/logics/morphoblue/configs.ts (hunk 10 above)
Comment (LuPoYi) [reply src/logics/morphoblue/configs.ts:8]: fixed

File: src/logics/compound-v3/tokens/data/arbitrum.json (hunk 6 above)
Comment (LuPoYi) [reply src/logics/compound-v3/tokens/data/arbitrum.json]: fixed

File: src/logics/compound-v3/tokens/data/arbitrum.json (hunk 5 above)
Comment (LuPoYi) [reply src/logics/compound-v3/tokens/data/arbitrum.json]: fixed

File: src/logics/compound-v3/tokens/data/arbitrum.json (hunk 4 above)
Comment (LuPoYi) [reply src/logics/compound-v3/tokens/data/arbitrum.json]: fixed

File: src/logics/morphoblue/configs.ts (hunk 11 above)
Comment (LuPoYi) [reply src/logics/morphoblue/configs.ts:9]: fixed

**Review by Jeff-CCH — COMMENTED** [review]
LGTM

**Review by Jeff-CCH — APPROVED** [review]

**Pull request merged**